packages/cozy-stack-client/src/DocumentCollection.js

```diff
@@ -468,9 +479,14 @@ class DocumentCollection {
           sort.push({ [field]: sortOrder })
       }
     }
+    // We need to pass the partialFilter in the selector, otherwise CouchDB might
+    // fallback on another index if it does not exist yet
+    const mergedSelector = partialFilter
+      ? merge(selector, partialFilter)
+      : selector
 
     const opts = {
-      selector,
+      selector: mergedSelector,
```

Comment:
Can you explain why this is needed? Since now we return an error if the index is not correct, Cozy Client will stop its flow, create the correct index and retry the request. Why do we need to pass to partial filter to the selector?

Reply:
This is not required per se, but the CouchDB doc suggests to add the content of the partial index in the selector, so it seems worth adding it: https://docs.couchdb.org/en/stable/api/database/find.html#find-partial-indexes

Reply:
🤷‍♂️